src/panels/lovelace/cards/hui-map-card.js

```diff
@@ -0,0 +1,258 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+import Leaflet from 'leaflet';
+
+import '../../map/ha-entity-marker.js';
+
+import setupLeafletMap from '../../../common/dom/setup-leaflet-map.js';
+import processConfigEntities from '../common/process-config-entities.js';
+import computeStateDomain from '../../../common/entity/compute_state_domain.js';
+import computeStateName from '../../../common/entity/compute_state_name.js';
+import debounce from '../../../common/util/debounce.js';
+
+Leaflet.Icon.Default.imagePath = '/static/images/leaflet';
+
+class HuiMapCard extends PolymerElement {
+  static get template() {
+    return html`
+      <style>
+        :host([is-panel]) ha-card {
+          left: 0;
+          top: 0;
+          width: 100%;
+          /**
+           * In panel mode we want a full height map. Since parent #view 
+           * only sets min-height, we need absolute positioning here
+           */
+          height: 100%;
+          position: absolute;
+        }
+        
+        #map {
+          z-index: 0;
+          border: none;
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 100%;
+          height: 100%;
+        }
+        
+        #root {
+          width: 100%;
+          position: relative;
+        }
+        
+        :host([is-panel]) #root {
+          height: 100%;
+        }
+      </style>
+      
+      <ha-card id="card" header="[[_config.title]]">
+        <div id="root">
```

Comment:
Can't we merge the logic of root into ha-card?

Reply:
Yes we could, but then if a title is used, `aspect_ratio: 100%` gives you a square card instead of a square map. I'm not sure which is preferable

Reply:
ah ok, let's keep it as is.